rclcpp/include/rclcpp/publisher.hpp

```diff
@@ -158,11 +157,13 @@ class Publisher : public PublisherBase
       // Register the publisher with the intra process manager.
       if (qos.history() != rclcpp::HistoryPolicy::KeepLast) {
         throw std::invalid_argument(
-                "intraprocess communication allowed only with keep last history qos policy");
+                "intraprocess communication on topic " + topic +
```

Comment:
Small nit, can we put the topic name in single quotes or brackets so that it is visibly separate from rest of the log message?  Especially since it's in the middle.

Reply:
also, shouldn't we use `get_topic_name()` here to get the full resolved topic name? `topic` is strictly what the user provides. Same below and for subscriptions.

Reply:
I think using the input `topic` is the most sensible since that is what they specified, so its easier inline in the code to see where the mistake was made.